pkg/virt-handler/vm.go

```diff
@@ -21,6 +22,18 @@ func NewVMController(listWatcher cache.ListerWatcher, domainManager libvirt.Doma
 				recorder.Event(vm, kubev1.EventTypeWarning, v1.SyncFailed.String(), err.Error())
 				return cache.ErrRequeue{Err: err}
 			}
+
+			restClient, err := kubecli.GetRESTClient()
```

Comment:
Could you also add a unit test?

If you fetch the rest client this way, it is hard to write tests.